torchtune/config/_utils.py

```diff
@@ -148,3 +150,37 @@ def _merge_yaml_and_cli_args(yaml_args: Namespace, cli_args: List[str]) -> DictC
 
     # CLI takes precedence over yaml args
     return OmegaConf.merge(yaml_conf, cli_conf)
+
+
+def _get_template(template: str) -> PromptTemplate:
+    """
+    Get the prompt template class from the template string.
+
+    String should either be the PromptTemplate class name directly, or a raw
+    string with 1 or more placeholders. If none of these apply, then raise an
```

Comment:
Dumb q: how often do people actually want to configure prompt template by directly passing a format string? E.g. 

```
tune full_finetune_single_device --config llama2/7B_full dataset.template=f"Below is an instruction that describes a task, paired with an input that provides further context. Write a response that appropriately completes the request.\n\n### Instruction:\n{instruction}\n\n### Input:\n{input}\n\n### Response:\n"
```

doesn't exactly roll off the keyboard

Reply:
Also I guess a higher-level point here: is the support for configurable f strings worth the complexity that we are adding? This is a bunch of stuff happening on the config backend basically just to have our config support a Union type. Wondering if there's a simpler way we can expose the f-string functionality (e.g. provide a builder taking in f-string as its only arg)

Reply:
> doesn't exactly roll off the keyboard

This could be specified in the config file itself, which I can see being a common use case

> Wondering if there's a simpler way we can expose the f-string functionality (e.g. provide a builder taking in f-string as its only arg)

yes I was planning to have a CustomTemplate class that does exactly this, but you'd still need to check to see if a string was passed instead of a template name

Reply:
> yes I was planning to have a CustomTemplate class that does exactly this, but you'd still need to check to see if a string was passed instead of a template name

Just have 

```
CustomTemplate(PromptTemplate):
	def __init__(self, f_string):
		self.template = f_string
	
	def format()...
```
and either 

```
template:
	_component_: AlpacaInstructTemplate
```
or
```
template:
	_component_: CustomTemplate
	f_string: "{user}: a, {system}: b"
```

Then `instantiate(cfg.template)` will work either way, no? (I might be making a stupid config mistake here, lmk if so)

Reply:
the template goes in the builder function, so it would be a nested component :) this was the reason I needed a str mapping here unfortunately... I don't like it but I haven't figured out a workaround

Reply:
I thought we can get around nested components by building separately and piecing together in the recipe? Anyways if you think that hurts UX too much (or I am still missing your point) feel free to ignore